services/url-discovery-service/src/url_discovery_service/main.py

```
@@ -0,0 +1,180 @@
+    allow_credentials=True,
+    allow_methods=["*"],
+    allow_headers=["*"],
+)
+
+
+def is_same_domain(url1: str, url2: str) -> bool:
+    """Check if two URLs are from the same domain."""
+    try:
+        domain1 = urlparse(url1).netloc
+        domain2 = urlparse(url2).netloc
+        return domain1 == domain2
```

Ruff (0.14.0)

Consider moving this statement to an `else` block

(TRY300)